tests/Unit/Parallel/Test_GlobalCache.cpp

```diff
@@ -228,13 +228,24 @@ class UseCkCallbackAsCallback : public Parallel::Callback {
   void pup(PUP::er& /*p*/) override {
     ERROR(
         "Should not be pupping a UseCkCallbackAsCallback, since it is always "
-        "stored in the local MutableGlobalCache");
+        "stored in the local GlobalCache");
   }
 
  private:
   CkCallback callback_;
 };
 
+// We have to define our expected mutable values statically out here because the
+// mutable part of the cache is once per node and we are only running on one
+// node. This means that when we run `run_test_one` on the first array element
+// and it mutates a mutable tag, that tag gets immediately changed inside the
+// mutable part of the cache. Then when the second array element runs
+// `run_test_one` this value has already been mutated, so we need to keep track
+// of these changes outside the functions with these static vars.
+static double expected_weight = 160.0;
+static std::string expected_email = "joe@somewhere.com";
+static size_t expected_number_of_legs = 6;
```

Comment:
Can you explain in a github comment what is different from the old behavior that requires this?

Reply:
So I originally left this part of the test alone, but found that it failed. Here is the flow of the `run_test_one` function:

1. Check value of tag is expected *starting* value in `run_test_one`
2. Mutate tag in `run_test_one`
3. Immediately call `run_test_two` inlined.

Now this is run on every element of the array parallel component (4 elements). What this means is that the mutable tag will be edited 4 times because `run_test_one` is run on all array elements (not a big deal because it's being mutated to the same value). It appears that in the old test, possibly because the mutate went through a proxy to the const cache and then a proxy to the mutable cache, the actual mutate call wasn't run until all 4 `run_test_one` functions were called. This means that step 1., the check that the tag is its *starting* value, always passed. However, now with my changes the mutate only goes through one proxy call to the cache. Apparently this caused the actual mutate to be run before all 4 of the `run_test_one` functions could run, so step 1. failed in at least one `run_test_one` because the tag is no longer at its starting value.

That's why I added the static variables. To keep track of the actual current state of the tags so that they can be compared in all `run_test_X` whenever they are run. I think the test before just got lucky that that's how the execution order worked out.

Reply:
I think that breaks a lot of the purpose of the test.  All the `mutable_cache_item_is_ready` calls are trivially being satisfied if everything is inline, right?  I don't think this is testing registering and calling callbacks anymore.

It might be necessary to have the different threads do different things to make them they actually have to wait on one another.

Reply:
What if I leave the functions as they originally were except I remove the calls to the other functions inside them. Then I call each one on a different element? Like this:

```cpp
array_proxy[3].run_test_five();
array_proxy[2].run_test_four();
array_proxy[1].run_test_three();
array_proxy[0].run_test_two();
array_proxy[3].run_test_one();
```

Since `run_test_one` is what does the mutate that kicks everything off, a bunch of callbacks should be registered and called.

Reply:
Yes, that sounds good.  I don't see any reason they have to be chained.